### PR TITLE
Provide Azure Managed Identity authorization foundation

### PR DESCRIPTION
**Pull request by tomkerkhove**
Provide Azure Managed Identity authorization which can be used with services such as Azure API Management.

Relates to https://github.com/arcus-azure/arcus.webapi/issues/120

### CODE REVIEW

File: src/Arcus.WebApi.Security.Authorization/Arcus.WebApi.Security.Authorization.csproj [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+      <TargetFramework>netstandard2.0</TargetFramework>
+      <Authors>Arcus</Authors>
+      <Company>Arcus</Company>
+      <RepositoryType>Git</RepositoryType>
+      <PackageTags>Azure;WebAPI;App Services;Web App;Web;API</PackageTags>
+      <Description>Provides capabilities to easily build secure Web APIs running in Azure.</Description>
```
Comment (tomkerkhove) [inline src/Arcus.WebApi.Security.Authorization/Arcus.WebApi.Security.Authorization.csproj]: I'll update this to reflect what the package does

**Comment by tomkerkhove** [comment]
I was going to do it in one PR but agree that it's best to merge this.
Ok for you as well @stijnmoreels?

I can create a preview release and run it at the customer. MyGet package is not an option unfortunately since we're using containers and other NuGet feeds are a pain.

**Comment by stijnmoreels** [comment]
> I was going to do it in one PR but agree that it's best to merge this.
> Ok for you as well @stijnmoreels?
> 
> I can create a preview release and run it at the customer. MyGet package is not an option unfortunately since we're using containers and other NuGet feeds are a pain.

Hmm, I've it's not too much, I would like to keep func/tests together and do it in this PR, no? Not sure what the benifit would be of merging it already?

**Comment by tomkerkhove** [comment]
To start using it early and see if it's decent from a user perspective.

For testing we'll need to wire up APIM or so most probably

**Comment by stijnmoreels** [comment]
> To start using it early and see if it's decent from a user perspective.
> 
> For testing we'll need to wire up APIM or so most probably

Seems a bit quick and dirty to do that on master but for now it's OK I guess.

**Comment by tomkerkhove** [comment]
I've added basic tests for now @stijnmoreels, don't worry - More will follow.

**Pull request merged**